### DISCUSSION TITLE
If a property in the source list does not exists its value is set null in the destination

### DISCUSSION THREAD

**Discussion by cam92yz** [LuckyPennySoftware/AutoMapper#3968] [NA]
Hello

We updated AutoMapper to version 11. After the upgrade we faced the problem that the AutoMapper sets a property to null if it does not exist in the source class. The behavior was only recognized in lists.

`        

        var mapper = new AutoMapper.Mapper(mapperConfig);

        var destination = new Destination();
        destination.Group = new Group();
        destination.Definitions = new List<Definition>();
        destination.Definitions.Add(new Definition() { Group = new Group() });

        var source = new Source();
        source.Definitions = new List<BusinessDefinition>();
        source.Definitions.Add(new BusinessDefinition());
        mapper.Map(source, destination);

        var x = destination.Definitions[0].Group; // The property does not exist on the BusinessDefinition class. After mapping the "Group" property is null, i expect that the property keeps its value. 
        var z = destination.Group; // This property does not exist in the "Source" class. After mapping the value did not change in this case.
`

[ConsoleApp7.zip](https://github.com/AutoMapper/AutoMapper/files/8603822/ConsoleApp7.zip)

**Comment by lbargaoanu** [comment]
https://docs.automapper.org/en/latest/11.0-Upgrade-Guide.html#mapping-into-existing-collections

**Reply by cam92yz** [reply]
Thank you

It might be, that i understand something wrong from the upgrade guide, but my collections are not readonly therefore i think that is not
the problem.

`    
    

    public class Destination
    {
        public Group Group { get; set; }
        public IList<Definition> Definitions { get; set; }
    }

    public class Definition
    {
        public Group Group { get; set; }
    }

    public class Group
    {
        public string Name { get; set; } = "blaaaaa";
    }

    public class Source
    {
        public IList<BusinessDefinition> Definitions { get; set; }
    }

    public class BusinessDefinition
    {
    }

`

**Reply by lbargaoanu** [reply]
Then your code has nothing to do with the 11 release. Whatever it is, 10.1.1 does the same, and you should just fix your code.

**Reply by cam92yz** [reply]
You are right. Although the described behavior is strange. It's my mistake, i noticed that nothing chaged with the Automapper. The problem that i had in the code, looked like that something has changed with the Automapper.

Thanks

You can close it.
